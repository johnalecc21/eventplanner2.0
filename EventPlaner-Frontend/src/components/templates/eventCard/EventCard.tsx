import { Calendar, MapPin } from 'lucide-react';

interface EventCardProps {
  title: string;
  image: string;
  date: string;
  location: string;
  category: string;
}

const EventCard = ({ title, image, date, location, category }: EventCardProps) => {
  return (
    <div
      className="glass-effect rounded-xl overflow-hidden group cursor-pointer shadow-lg transition-shadow duration-300 hover:shadow-2xl"
      style={{ background: 'var(--background)', color: 'var(--text)' }}
      data-testid="event-card"
    >
      <div className="relative h-48 overflow-hidden" data-testid="event-image-container">
        <img
          src={image}
          alt={title}
          className="w-full h-full object-cover group-hover:scale-110 transition-transform duration-300"
          data-testid="event-image"
        />
        <div
          className="absolute top-4 right-4 px-3 py-1 rounded-full glass-effect text-sm font-medium"
          style={{ background: 'var(--primary)', color: '#fff' }}
          data-testid="event-category"
        >
          {category}
        </div>
      </div>
      <div className="p-6 space-y-4" style={{ background: 'var(--background)' }} data-testid="event-content">
        <h3 className="text-xl font-semibold" style={{ color: 'var(--text)' }} data-testid="event-title">
          {title}
        </h3>
        <div className="space-y-2" data-testid="event-details">
          <div className="flex items-center space-x-2" data-testid="event-date">
            <Calendar className="h-4 w-4" style={{ color: 'var(--accent)' }} />
            <span className="text-sm" style={{ color: 'var(--text-secondary)' }}>
              {date}
            </span>
          </div>
          <div className="flex items-center space-x-2" data-testid="event-location">
            <MapPin className="h-4 w-4" style={{ color: 'var(--accent)' }} />
            <span className="text-sm" style={{ color: 'var(--text-secondary)' }}>
              {location}
            </span>
          </div>
        </div>
      </div>
    </div>
  );
};

export default EventCard;