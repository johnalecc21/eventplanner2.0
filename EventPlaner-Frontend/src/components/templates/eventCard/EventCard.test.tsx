// EventCard.test.tsx
import { render, screen } from '@testing-library/react';
import EventCard from './EventCard';

describe('EventCard Component', () => {
  const mockProps = {
    title: 'Music Festival',
    image: 'https://via.placeholder.com/150',
    date: 'July 25, 2024',
    location: 'New York City',
    category: 'Concert',
  };

  test('renders the EventCard component', () => {
    render(<EventCard {...mockProps} />);
    expect(screen.getByTestId('event-card')).toBeInTheDocument();
  });

  test('displays the event title', () => {
    render(<EventCard {...mockProps} />);
    expect(screen.getByTestId('event-title')).toHaveTextContent(mockProps.title);
  });

  test('displays the event image', () => {
    render(<EventCard {...mockProps} />);
    const imageElement = screen.getByTestId('event-image');
    expect(imageElement).toHaveAttribute('src', mockProps.image);
    expect(imageElement).toHaveAttribute('alt', mockProps.title);
  });

  test('displays the event category', () => {
    render(<EventCard {...mockProps} />);
    expect(screen.getByTestId('event-category')).toHaveTextContent(mockProps.category);
  });

  test('displays the event date', () => {
    render(<EventCard {...mockProps} />);
    expect(screen.getByTestId('event-date')).toHaveTextContent(mockProps.date);
  });

  test('displays the event location', () => {
    render(<EventCard {...mockProps} />);
    expect(screen.getByTestId('event-location')).toHaveTextContent(mockProps.location);
  });

  test('matches snapshot', () => {
    const { asFragment } = render(<EventCard {...mockProps} />);
    expect(asFragment()).toMatchSnapshot();
  });
});
