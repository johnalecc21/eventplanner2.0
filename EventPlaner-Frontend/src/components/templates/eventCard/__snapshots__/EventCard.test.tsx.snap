// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`EventCard Component matches snapshot 1`] = `
<DocumentFragment>
  <div
    class="glass-effect rounded-xl overflow-hidden group cursor-pointer shadow-lg transition-shadow duration-300 hover:shadow-2xl"
    data-testid="event-card"
  >
    <div
      class="relative h-48 overflow-hidden"
      data-testid="event-image-container"
    >
      <img
        alt="Music Festival"
        class="w-full h-full object-cover group-hover:scale-110 transition-transform duration-300"
        data-testid="event-image"
        src="https://via.placeholder.com/150"
      />
      <div
        class="absolute top-4 right-4 px-3 py-1 rounded-full glass-effect text-sm font-medium"
        data-testid="event-category"
        style="color: rgb(255, 255, 255);"
      >
        Concert
      </div>
    </div>
    <div
      class="p-6 space-y-4"
      data-testid="event-content"
    >
      <h3
        class="text-xl font-semibold"
        data-testid="event-title"
      >
        Music Festival
      </h3>
      <div
        class="space-y-2"
        data-testid="event-details"
      >
        <div
          class="flex items-center space-x-2"
          data-testid="event-date"
        >
          <svg
            class="lucide lucide-calendar h-4 w-4"
            fill="none"
            height="24"
            stroke="currentColor"
            stroke-linecap="round"
            stroke-linejoin="round"
            stroke-width="2"
            viewBox="0 0 24 24"
            width="24"
            xmlns="http://www.w3.org/2000/svg"
          >
            <path
              d="M8 2v4"
            />
            <path
              d="M16 2v4"
            />
            <rect
              height="18"
              rx="2"
              width="18"
              x="3"
              y="4"
            />
            <path
              d="M3 10h18"
            />
          </svg>
          <span
            class="text-sm"
          >
            July 25, 2024
          </span>
        </div>
        <div
          class="flex items-center space-x-2"
          data-testid="event-location"
        >
          <svg
            class="lucide lucide-map-pin h-4 w-4"
            fill="none"
            height="24"
            stroke="currentColor"
            stroke-linecap="round"
            stroke-linejoin="round"
            stroke-width="2"
            viewBox="0 0 24 24"
            width="24"
            xmlns="http://www.w3.org/2000/svg"
          >
            <path
              d="M20 10c0 6-8 12-8 12s-8-6-8-12a8 8 0 0 1 16 0Z"
            />
            <circle
              cx="12"
              cy="10"
              r="3"
            />
          </svg>
          <span
            class="text-sm"
          >
            New York City
          </span>
        </div>
      </div>
    </div>
  </div>
</DocumentFragment>
`;
